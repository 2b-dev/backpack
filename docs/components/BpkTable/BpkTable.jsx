import React from 'react'
import CssModules from 'react-css-modules'

import styles from './bpk-table.scss'

const BpkTable = (props) => <table styleName='bpk-table' {...props} />

export default CssModules(BpkTable, styles)
