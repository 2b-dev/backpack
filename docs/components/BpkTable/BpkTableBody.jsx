import React from 'react'
import CssModules from 'react-css-modules'

import styles from './bpk-table.scss'

const BpkTableBody = (props) => <tbody {...props} />

export default CssModules(BpkTableBody, styles)
