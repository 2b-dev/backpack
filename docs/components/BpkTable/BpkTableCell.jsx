import React from 'react'
import CssModules from 'react-css-modules'

import styles from './bpk-table.scss'

const BpkTableCell = (props) => <td styleName='bpk-table__cell' {...props} />

export default CssModules(BpkTableCell, styles)
