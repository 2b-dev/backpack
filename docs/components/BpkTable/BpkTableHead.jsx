import React from 'react'
import CssModules from 'react-css-modules'

import styles from './bpk-table.scss'

const BpkTableHead = (props) => <thead {...props} />

export default CssModules(BpkTableHead, styles)
