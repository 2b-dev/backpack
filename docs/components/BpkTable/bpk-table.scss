@import './../../../scss/bonds';

.bpk-table {
  width: 100%;
  margin-bottom: $bpk-spacing-base;
  border-collapse: collapse;
  box-shadow: 0 0 0 1px $bpk-color-gray-100;
  table-layout: fixed;

  &__cell {
    padding: $bpk-spacing-sm;

    &--head {
      background-color: $bpk-color-gray-50;
      font-weight: normal;
      text-align: left;
    }
  }
}
