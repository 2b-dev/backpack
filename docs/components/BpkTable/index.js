import table from './BpkTable.jsx'
import tHead from './BpkTableHead.jsx'
import tBody from './BpkTableBody.jsx'
import tr from './BpkTableRow.jsx'
import th from './BpkTableHeadCell.jsx'
import td from './BpkTableCell.jsx'

export const BpkTable = table
export const BpkTableHead = tHead
export const BpkTableBody = tBody
export const BpkTableRow = tr
export const BpkTableCell = td
export const BpkTableHeadCell = th
