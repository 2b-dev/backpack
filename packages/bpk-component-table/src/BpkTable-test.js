jest.unmock('./BpkTable.jsx')

import React from 'react'
import { shallow } from 'enzyme'
import BpkTable from './BpkTable.jsx'

describe('BpkTable', () => {
  it('should render the correct element', () => {
    const wrapper = shallow(
      <BpkTable>
        <tbody />
      </BpkTable>
    )
    expect(wrapper.name()).toEqual('table')
  })

  it('should render the correct class', () => {
    const wrapper = shallow(
      <BpkTable>
        <tbody />
      </BpkTable>
    )
    expect(wrapper.prop('className')).toEqual('bpk-table')
  })
})
