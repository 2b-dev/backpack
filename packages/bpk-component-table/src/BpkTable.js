import PropTypes from 'prop-types';
import React from 'react';

import './bpk-table.scss';

const BpkTable = props => <table className="bpk-table">{props.children}</table>;

BpkTable.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BpkTable;
