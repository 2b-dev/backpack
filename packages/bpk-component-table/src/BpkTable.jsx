import React, { PropTypes } from 'react'

import './bpk-table.scss'

const BpkTable = ({ children }) => <table className='bpk-table' children={children} />

BpkTable.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default BpkTable
