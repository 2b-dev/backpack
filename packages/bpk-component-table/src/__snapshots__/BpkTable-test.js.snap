exports[`BpkTable should render correctly 1`] = `
<table
  className="bpk-table">
  <tbody />
</table>
`;
